### PR TITLE
Feature/user attendances

### PR DESCRIPTION
**Pull request by adanvillarreal**
### What does this PR do?

* Adds index for user's attendances
* Allows users to select a date range to view their total attendances and weekly frequency
* Includes gems for charts

#### Merge Checklist:

- [x] The branch is up-to-date (i.e. rebased) with the base branch
- [x] All tests are passing - see ["Running tests"](../#running-tests)
- [x] No code style issues

### CODE REVIEW

File: app/controllers/attendances_controller.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+class AttendancesController < ApplicationController
+  before_action :set_attendance, only: [:show, :edit, :update, :destroy]
+
+  # GET /attendances
+  # GET /attendances.json
+  def index
+    # @select_options = {
```
Comment (HectorMg) [inline app/controllers/attendances_controller.rb]: Should we remove this commented block

File: app/controllers/attendances_controller.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if @end_date.nil?
+      @end_date = Time.now.end_of_week
+    else
+      @end_date = Time.strptime(@end_date, "%m/%d/%Y")
+    end
+
+    @attendances = current_user.attendances
+    @attendances = @attendances.group_by_period(:day_of_week, :date, range: @start_date..@end_date, format: "%a").count
+    puts @attendances
+    @attendances_count = @attendances.sum{ |k, v| v }
```
Comment (HectorMg) [inline app/controllers/attendances_controller.rb]: This method is way too long, let's refactor what we can out to different methods please!

File: app/controllers/attendances_controller.rb (hunk 1 above)
Comment (adanvillarreal) [reply app/controllers/attendances_controller.rb]: done.

File: app/controllers/attendances_controller.rb (hunk 2 above)
Comment (adanvillarreal) [reply app/controllers/attendances_controller.rb]: Just created method for `@start_date` and `@end_date` initialization

**Pull request merged**